src/networking.c

```diff
@@ -2778,7 +2876,7 @@ void processMultibulkBuffer(client *c) {
         serverAssertWithInfo(c, NULL, c->argc == 0);
 
         /* Multi bulk length cannot be read without a \r\n */
-        newline = strchr(c->querybuf + c->qb_pos, '\r');
+        newline = memchr(c->querybuf + c->qb_pos, '\r', sdslen(c->querybuf) - c->qb_pos);
```

Comment:
Was this related to to the stopping condition that was mentioned? strchr should always work, because c->querybuf should always be null terminated.

Reply:
For some reason in dual-channel testswith ASan, we're experiencing timeout issues when the querybuf reaches ~4MB. The problem occurs because ASan takes an unusually long time to execute `strchr` operations, which run four times per command on a buffer filled with short set commands. While ASan doesn't report any errors, it takes approximately 10 minutes to process the querybuf. I know ASan implements a special interceptor for `strchr` and it may be slow see similar issue with `strstr` https://github.com/google/sanitizers/issues/1832, but I haven't yet found why this issue only happens when the I/O thread allocates the buffer. Regardless I believe `memchr` is more safer.